### PR TITLE
[Ruby] Adding known Azure Environments in ruby runtime for easy discovery

### PR DESCRIPTION
**Pull request by vishrutshah**
NodeJS Reference: https://github.com/Azure/autorest/blob/master/src/client/NodeJS/ms-rest-azure/lib/azureEnvironment.js

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vishrutshah__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/azure_environment.rb [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      attr_reader :active_directory_resource_id
+
+      # @return [String] the Active Directory resource ID
+      attr_reader :active_directory_graph_resource_id
+
+      # @return [String] the Active Directory resource ID
+      attr_reader :active_directory_graph_api_version
+
+      # @return [String] the endpoint suffix for storage accounts
+      attr_reader :storage_endpoing_suffix
```
Comment (veronicagg) [inline src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/azure_environment.rb]: typo: endpoint

**Comment by veronicagg** [comment]
@vishrutshah Left a minor comment, rest looks good to me. Per our conversation yesterday, we should update https://github.com/Azure/autorest/blob/master/src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/active_directory_service_settings.rb as well. Thanks!

File: src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/azure_environment.rb (hunk 1 above)
Comment (vishrutshah) [inline src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/azure_environment.rb]: Thanks for catching. Corrected.

**Comment by vishrutshah** [comment]
Yup. Updated !!

File: src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/active_directory_service_settings.rb [hunk 2]
```diff
@@ -19,20 +19,44 @@ class ActiveDirectoryServiceSettings
     #
     # @return [ActiveDirectoryServiceSettings] settings required for authentication.
     def self.get_azure_settings
-      settings = ActiveDirectoryServiceSettings.new
-      settings.authentication_endpoint = 'https://login.windows.net/'
```
Comment (veronicagg) [inline src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/active_directory_service_settings.rb]: This value seems to be different with the new settings:  'https://login.microsoftonline.com/', which one should it be?

Comment (vishrutshah) [inline src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/active_directory_service_settings.rb]: Yes https://login.microsoftonline.com/ one is new endpoint that we should use. The one that we were using (https://login.windows.net/) is the old endpoint and as per my conversation with @amarzavery it's for backward compatibility.

**Comment by veronicagg** [comment]
LGTM!

**Comment by vishrutshah** [comment]
@veronicagg Thanks for the review! As per our offline talk, i have used the old endpoint for now. I'll make another PR to be merged in just before releasing ruby runtime to give us time to fix recordings.

**Pull request merged**